### PR TITLE
Implement V16 Config class

### PR DESCRIPTION
**Pull request by david-allison-1**
This was the final class we needed for the V16 conversion

## Fixes
Related #8988

## How Has This Been Tested?

Unit Tests only

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java [hunk 1]
```diff
@@ -2243,7 +2243,7 @@ private int _rolloverHour() {
 
     @Override
     public boolean _new_timezone_enabled() {
-        return getCol().has_config_not_null("creationOffset");
+        return getCol().has_config("creationOffset");
```
Comment (dae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java:2246]: Not sure this is safe - the "exists but is null" case is treated as disabled by sched_timing_today(), and it starts off as null in the schema11 vars by the looks of things.

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java:2246]: Thanks! Was unset in the AnkiDroid.

is `GetAllConfig` -> `check for null` the best way to go about this here?

Comment (dae) [reply AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java]: Couldn't you use GetConfigJson() and treat NotFound/null/anything except true as false? That'd save you having to grab all the config at once.

**Review by mikehardy — APPROVED** [review]
Going to trust between you and Damien you've thought through this - it looks okay to me at least (assuming we don't need a new RustBackend build / aar commit to match?) and I'm fine with merging at will to keep the process moving - sounds like we're getting close here and it will all be exercised for real soon enough...

**Pull request merged**